src/controls/controls.actions.ts

```diff
@@ -268,134 +269,80 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
   }
 
   /**
-   * Detect if the fabric object is flipped on one side.
-   * @param {fabric.Object} target
-   * @return {Boolean} true if one flip, but not two.
+   * @see https://github.com/fabricjs/fabric.js/pull/8380
    */
-  function targetHasOneFlip(target) {
-    return target.flipX !== target.flipY;
-  }
-
-  /**
-   * Utility function to compensate the scale factor when skew is applied on both axes
-   * @private
-   */
-  function compensateScaleForSkew(
-    target,
-    oppositeSkew,
-    scaleToCompensate,
-    axis,
-    reference
+  function skewObject(
+    axis: 'x' | 'y',
+    { target, ex, ey, originX, originY, skewX, skewY, flipX, flipY },
+    pointer: Point
   ) {
-    if (target[oppositeSkew] !== 0) {
-      var newDim = target._getTransformedDimensions()[axis];
-      var newValue = (reference / newDim) * target[scaleToCompensate];
-      target.set(scaleToCompensate, newValue);
+    const { otherAxis, otherScaleKey, counterOrigin, key, valueBefore } =
+        axis === 'y'
+          ? {
+              otherAxis: 'x',
+              otherScaleKey: 'scaleX',
+              counterOrigin: target.resolveOriginX(originX),
+              key: 'skewY',
+              valueBefore: skewY,
+            }
+          : {
+              otherAxis: 'y',
+              otherScaleKey: 'scaleY',
+              counterOrigin: target.resolveOriginY(originY),
+              key: 'skewX',
+              valueBefore: skewX,
+            },
+      dim = target._getTransformedDimensions(),
+      offset = pointer.subtract(new Point(ex, ey))[axis],
```

Comment:
I will try

Reply:
Tried it and doesn't seem to do any good.
Seems accurate to the point except when crossing over to negative values
Correct me if I am wrong

Reply:
Although it seems that the problem is when it crosses to negative values, but in reality it is because of the arrangement of the polygon points.

That's the problem I mentioned [here.](https://github.com/fabricjs/fabric.js/pull/8380#issuecomment-1283383318) Depending on the arrangement of the polygon points, a change in `skew` will not necessarily change the `width` of the bbox (or it will not vary the `width` in the same proportion as the mouse movement)

I made an example with two cases that will be clearer: https://codesandbox.io/s/skew-control-0frlui?file=/src/index.js 

Try applying `skewX` to the rectangle. See that it works correctly, with good accuracy even crossing to negative values ​​to some extent.

![control skew square](https://user-images.githubusercontent.com/65685842/196820861-ab8ccc83-1eae-4f84-82e9-ca7b57639367.gif)

Then try to apply `skewX` on the triangle. The mouse movement does not follow the control point. Because the triangle only starts to change its width when the top vertex reaches one of the edges.

![control skew triangle](https://user-images.githubusercontent.com/65685842/196821206-1178e18d-aa3f-45e4-8b03-1ea78f4105dc.gif)

So depending on the arrangement of the points the mouse movement will follow the control point well or not. (I don't see this as a big problem)

Reply:
If you think that for UX it is really important for the mouse to be fully synchronized with the control point you can compensate for the difference with the scale. For example, you moved the mouse to the right 10 pixels, but the polygon after applying the new skewX grew only 5 pixels wide, you apply a new scaleX factor to reach 10 pixels and stay in sync with the mouse.